VideoLocker/src/main/java/org/edx/mobile/view/DiscussionAddPostFragment.java

```diff
@@ -234,18 +234,26 @@ protected void getTopicList() {
             @Override
             public void onSuccess(CourseTopics courseTopics) {
                 final ArrayList<DiscussionTopic> allTopics = new ArrayList<>();
-                allTopics.addAll(courseTopics.getCoursewareTopics());
                 allTopics.addAll(courseTopics.getNonCoursewareTopics());
+                allTopics.addAll(courseTopics.getCoursewareTopics());
 
                 final TopicSpinnerAdapter adapter = new TopicSpinnerAdapter(container.getContext(), DiscussionTopicDepth.createFromDiscussionTopics(allTopics));
                 topicsSpinner.setAdapter(adapter);
 
                 {
                     // Attempt to select the topic that we navigated from
-                    // Otherwise, leave the default option, which is "Choose a topic..."
-                    int selectedTopicIndex = adapter.getPosition(discussionTopic);
-                    if (selectedTopicIndex >= 0) {
-                        topicsSpinner.setSelection(selectedTopicIndex);
+                    // Otherwise, leave the default option, which is the first non-courseware topic
+                    if (!discussionTopic.isAllType() && !discussionTopic.isFollowingType()) {
+                        int selectedTopicIndex;
+                        if (discussionTopic.getIdentifier() == null) {
+                            // In case of a parent topic, we need to select the first child topic
+                            selectedTopicIndex = adapter.getPosition(discussionTopic.getChildren().get(0));
```

Comment:
We need a non-null and non-empty check on the children here; otherwise if we can't select top-level topics, then we should fallback to the default.